client/js/jsonedit.js

```diff
@@ -1475,16 +1483,59 @@ function jeSetAndSelect(replaceBy, insideString) {
   else
     var jsonString = JSON.stringify(jeStateNow, null, '  ');
   const startIndex = jsonString.indexOf(insideString ? '###SELECT ME###' : '"###SELECT ME###"');
-  let length = jsonString.length-15-(insideString ? 0 : 2);
-
+  const replaceByString = JSON.stringify(replaceBy);
   if(insideString)
-    jsonString = jsonString.replace(/###SELECT ME###/, JSON.stringify(replaceBy).substr(1, JSON.stringify(replaceBy).length-2));
+    jsonString = jsonString.replace(/###SELECT ME###/, replaceByString.substr(1, replaceByString.length-2));
   else
-    jsonString = jsonString.replace(/"###SELECT ME###"/, JSON.stringify(replaceBy));
+    jsonString = jsonString.replace(/"###SELECT ME###"/, replaceByString);
 
   jeSet(jsonString);
-  const quote = typeof replaceBy == 'string' && !insideString ? 1 : 0;
-  jeSelect(startIndex + quote, startIndex+jsonString.length-length - quote, true);
+
+  let bracket = 0;
+  let wavybracket = 0;
+  let begin = 0;
+  let end = 0;
+  if(replaceBy !== undefined){ 
+    const quoteArray = getAllIndexes(replaceByString,'\"');
+    if(quoteArray.length > 2) { // adjusts selection area to inside last set of quotes
+      begin = quoteArray[2] + 1;
+      end = quoteArray[3];
+    } else if (quoteArray.length > 0) {
+      begin = quoteArray[0] + 1;
+      end = quoteArray[1];
+    } else { // adjusts selection area to inside brackets
+      bracket = replaceByString.lastIndexOf('[') + 1;
+      wavybracket = replaceByString.lastIndexOf('{')+ 1;
+      begin = Math.max(bracket, wavybracket, 0);
+      bracket = replaceByString.lastIndexOf(']') > 0 ? replaceByString.lastIndexOf(']') > - 1 : replaceByString.length;
+      wavybracket = replaceByString.lastIndexOf('}') > 0 ? replaceByString.lastIndexOf('}') > - 1 : replaceByString.length;
+      end = Math.min(bracket, wavybracket);
+    }
+
+    // adjusts selection area for formatting of arrays and objects
+    if((replaceByString.indexOf('[') > -1 && (replaceByString.indexOf('\"') > -1 || replaceByString.indexOf('{') > -1)) || replaceByString.indexOf('\"') > -1 && replaceByString.indexOf('{') > -1) {
```

Comment:
`replaceByString.indexOf('{') > -1)) || replaceByString.indexOf('\"') > -1 && replaceByString.indexOf('{') > -1)` is identical to simply `replaceByString.indexOf('{') > -1)) `.

Reply:
I think that maybe was missing a set of parenthesis.  

(replaceByString.indexOf('[') > -1 && (replaceByString.indexOf('\"') > -1 || replaceByString.indexOf('{') > -1)) || (replaceByString.indexOf('\"') > -1 && replaceByString.indexOf('{') > -1))

It is supposed to evaluate for true if at least 2 out of the 3 `{, [, \"` are present. 

replaceByString.indexOf('[') > -1
replaceByString.indexOf('\"') > -1
replaceByString.indexOf('{') > -1)